bentoml/_internal/models/model.py

```diff
@@ -532,6 +546,36 @@ def with_options(self, **kwargs: t.Any) -> ModelInfo:
             creation_time=self.creation_time,
         )
 
+    # cached_property doesn't support __slots__ classes
+    @property
+    def imported_module(self) -> ModuleType:
```

Comment:
May be we can get rid of `self._cached_module` if we use a `@lru_cache(maxsize=1)` decorator here.

Reply:
i don't think that would work?

Reply:
I would think `lru_cache` also uses the dict, but will test this one out.

Reply:
Hm, after reading up on this `lru_cache` might not do exactly what we want; it'll cache externally over the `self` parameter without modifying the actual object so we'd have to either let it be unbounded, which would be a leak, or only cache a certain number.